api/v1/runtime_types.go

```diff
@@ -280,12 +276,20 @@ func (k *Runtime) UpdateStateDeletion(c RuntimeConditionType, r RuntimeCondition
 	meta.SetStatusCondition(&k.Status.Conditions, condition)
 }
 
-func (k *Runtime) UpdateStatePending(c RuntimeConditionType, r RuntimeConditionReason, status, msg string) {
-	if status == "False" {
-		k.Status.State = RuntimeStateFailed
-	} else {
-		k.Status.State = RuntimeStatePending
+func (k *Runtime) UpdateStateFailed(c RuntimeConditionType, r RuntimeConditionReason, msg string) {
+	k.Status.State = RuntimeStateFailed
+	condition := metav1.Condition{
+		Type:               string(c),
+		Status:             metav1.ConditionFalse,
+		LastTransitionTime: metav1.Now(),
+		Reason:             string(r),
+		Message:            msg,
 	}
+	meta.SetStatusCondition(&k.Status.Conditions, condition)
+}
+
+func (k *Runtime) UpdateStatePending(c RuntimeConditionType, r RuntimeConditionReason, status, msg string) {
```

Comment:
When we manipulate those methods I think we can also change third argument type (status) to  metav1.ConditionStatus everywhere eg: 

```go
UpdateStatePending(c RuntimeConditionType, r RuntimeConditionReason, status metav1.ConditionStatus, msg string)
UpdateStateDeletion(c RuntimeConditionType, r RuntimeConditionReason, status,  status metav1.ConditionStatus, msg string)
```